src/System/IO/Streams/Network/HAProxy.hs

```diff
@@ -281,8 +285,8 @@ parseNewHaProxy localProxyInfo = do
         void $ take $ fromIntegral nskip
 
         -- Note: we actually want the brain-dead constructors here
-        let sa = N.SockAddrInet (N.PortNum srcPort) srcAddr
-        let sb = N.SockAddrInet (N.PortNum destPort) destAddr
+        let sa = N.SockAddrInet (fromIntegral srcPort) srcAddr
```

Comment:
The `Num` instance for `PortNumber` is available since network-2.3 (lower bound of this package). `PortNum` constructor is now deprecated.